src/hooks/usePreferredMediaDevice.tsx

```diff
@@ -0,0 +1,31 @@
+import { t } from "i18next";
+import { useState } from "react";
+
+const usePreferredMediaDevice = () => {
+  const [preferredDeviceId, setPreferredDeviceId] = useState<string | null>(
+    () => {
+      if (typeof window !== "undefined") {
+        return localStorage.getItem("preferredDeviceId");
+      }
+      return null;
+    },
+  );
+
+  const setDeviceId = (deviceId: string) => {
+    if (typeof window !== "undefined") {
+      try {
+        localStorage.setItem("preferredDeviceId", deviceId);
```

Comment:
It's just setting to local storage, we don't need a try catch here. Remove the translation key as well.